apps/next-evans/pages/getting-started.tsx

```diff
@@ -0,0 +1,351 @@
+import React from 'react';
+import { PrismaClient } from '@prisma/client';
+import cx from 'classnames';
+import { t } from '@friends-library/locale';
+import { ChevronDownIcon } from '@heroicons/react/24/outline';
+import type { GetStaticProps } from 'next';
+import type { CoverProps } from '@friends-library/types';
+import Dual from '@/components/core/Dual';
+import Heading from '@/components/core/Heading';
+import BackgroundImage from '@/components/core/BackgroundImage';
+import BooksGrid from '@/public/images/explore-books.jpg';
+import DuoToneWaveBlock from '@/components/pages/getting-started/DuoToneWaveBlock';
+import EmbeddedAudio from '@/components/core/EmbeddedAudio';
+import { LANG } from '@/lib/env';
+import { makeScroller } from '@/lib/scroll';
+import GettingStartedPaths from '@/components/pages/getting-started/GettingStartedPaths';
+import recommendedBooks from '@/lib/recommended-books';
+import { mostModernEdition } from '@/lib/editions';
+import getCustomCode from '@/lib/get-custom-code';
+
+export const getStaticProps: GetStaticProps<Props> = async () => {
+  const prisma = new PrismaClient();
+
+  const friends = await prisma.friends.findMany({
+    where: { lang: LANG },
+    select: {
+      slug: true,
+      name: true,
+      documents: {
+        select: {
+          slug: true,
+          title: true,
+          editions: {
+            select: {
+              type: true,
+              edition_audios: {
+                select: {
+                  id: true,
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+  });
+
+  const customCode = await Promise.all(
+    friends.map((friend) =>
+      getCustomCode(
+        friend.slug,
+        friend.documents.map((doc) => doc.slug),
+      ),
+    ),
+  );
+
+  const allBooks: NeededCoverProps[] = friends.flatMap((friend, index) => {
+    const codeForFriend = customCode[index];
+    return friend.documents.map((doc) => {
+      const codeForDocument = codeForFriend ? codeForFriend[doc.slug] : undefined;
+      return {
+        authorSlug: friend.slug,
+        author: friend.name,
+        title: doc.title,
+        edition: mostModernEdition(doc.editions.map((edition) => edition.type)),
+        customCss: codeForDocument ? codeForDocument.css || `` : ``,
+        customHtml: codeForDocument ? codeForDocument.html || `` : ``,
+        hasAudio: doc.editions.some((edition) => edition.edition_audios),
+        slug: doc.slug,
+      };
+    });
+  });
+
+  function filterBooks(
```

Comment:
is this closing over anything from it's scope?  if not, let's move it out to the module level as a free standing function